core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1190,8 +1192,31 @@ class KafkaApis(val requestChannel: RequestChannel,
     val metadataRequest = request.body[MetadataRequest]
     val requestVersion = request.header.apiVersion
 
+    // Check if topicId is presented firstly.
+    val topicIds = metadataRequest.topicIds.asScala.toSet.filterNot(_ == Uuid.ZERO_UUID)
+    val supportedVersionTopicIds = if (config.interBrokerProtocolVersion >= KAFKA_2_8_IV1) topicIds else Set.empty[Uuid]
```

Comment:
@jolshan Shouldn't we use `KAFKA_2_8_IV1` for `KafkaConfig.usesTopicId` and use that here?

Reply:
Had a chat offline with Justine. Since `KafkaConfig.usesTopicId` is used only to decide whether to generate topic ids, 2_8_IV0 makes sense for that, and the check for KAFKA_2_8_IV1 here is the right one. Justine also mentioned:

> if we are in the process of bumping to 2_8_IV1, we could have a scenario where the controller is not yet bumped (and not sending topic IDs) but the broker serving the metadata is. In that case I think we would just get unknown topic ID errors and maybe that is ok. If the reverse is occurring, we could also have unsupported version exception when there are actually IDs.


Can we add tests for these two cases?

Reply:
Hello, I tried to add 2 test in `MetadataRequestBetweenDifferentIbpTest` to verify these 2 cases, this test will take probably 10 seconds since we need to constantly restart brokers to change the controller.